src/main/java/org/folio/dao/fund/FundPostgresDAO.java

```diff
@@ -2,26 +2,59 @@
 
 import static org.folio.rest.impl.FundAPI.FUND_TABLE;
 
+import java.util.Collections;
 import java.util.List;
 
+import io.vertx.ext.web.handler.HttpException;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.folio.rest.jaxrs.model.Fund;
 import org.folio.rest.persist.Criteria.Criterion;
 
 import io.vertx.core.Future;
+import org.folio.rest.persist.CriterionBuilder;
 import org.folio.rest.persist.DBConn;
 import org.folio.rest.persist.interfaces.Results;
 
+import javax.ws.rs.core.Response;
+
 public class FundPostgresDAO implements FundDAO {
+  private static final Logger logger = LogManager.getLogger();
+  private static final String FUND_NOT_FOUND = "Fund not found, id=%s";
 
-  private static final Logger logger = LogManager.getLogger(FundPostgresDAO.class);
+  @Override
+  public Future<Fund> getFundById(String id, DBConn conn) {
+    logger.debug("Trying to get fund by id {}", id);
+    return conn.getById(FUND_TABLE, id, Fund.class)
+      .map(fund -> {
+        if (fund == null) {
```

Comment:
getById gives null if it doesn't find a record?

Reply:
Yes. I would love to find the Javadoc for the Conn class, but in the meantime I look at the [implementation](https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java). The javadoc does not even say that it can return null, but you can see it in the code.